plenum/common/ledger.py

```diff
@@ -159,3 +159,10 @@ def get_last_committed_txn(self):
         if self.size > 0:
             return self.getBySeqNo(self.size)
         return None
+
+    def get_by_seq_no_uncommitted(self, seq_no):
+        if seq_no > self.uncommitted_size:
+            raise KeyError
```

Comment:
If we are okay with having `IndexError` insteead of `KeyError` in case `seq_no > self.uncommited_size` then this additional check is not needed.